sonar-db/src/test/java/org/sonar/db/version/CreateIndexBuilderTest.java

```diff
@@ -0,0 +1,145 @@
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.db.version;
+
+import java.util.List;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.sonar.db.dialect.H2;
+import org.sonar.db.dialect.MySql;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonar.db.version.VarcharColumnDef.newVarcharColumnDefBuilder;
+
+public class CreateIndexBuilderTest {
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Test
+  public void create_index_on_single_column() {
+    verifySql(new CreateIndexBuilder(new H2())
+      .setTable("issues")
+      .setName("issues_key")
+      .addColumn(newVarcharColumnDefBuilder().setColumnName("kee").setLimit(10).build()),
+      "CREATE INDEX issues_key ON issues (kee)");
+  }
+
+  @Test
+  public void create_unique_index_on_single_column() {
+    verifySql(new CreateIndexBuilder(new H2())
+      .setTable("issues")
+      .setName("issues_key")
+      .addColumn(newVarcharColumnDefBuilder().setColumnName("kee").setLimit(10).build())
+      .setUnique(true),
+      "CREATE UNIQUE INDEX issues_key ON issues (kee)");
+  }
+
+  @Test
+  public void create_index_on_multiple_columns() {
+    verifySql(new CreateIndexBuilder(new H2())
+      .setTable("rules")
+      .setName("rules_key")
+      .addColumn(newVarcharColumnDefBuilder().setColumnName("repository").setLimit(10).build())
+      .addColumn(newVarcharColumnDefBuilder().setColumnName("rule_key").setLimit(50).build()),
+      "CREATE INDEX rules_key ON rules (repository, rule_key)");
+  }
+
+  @Test
+  public void create_unique_index_on_multiple_columns() {
+    verifySql(new CreateIndexBuilder(new H2())
+      .setTable("rules")
+      .setName("rules_key")
+      .addColumn(newVarcharColumnDefBuilder().setColumnName("repository").setLimit(10).build())
+      .addColumn(newVarcharColumnDefBuilder().setColumnName("rule_key").setLimit(50).build())
+      .setUnique(true),
+      "CREATE UNIQUE INDEX rules_key ON rules (repository, rule_key)");
+  }
+
+  @Test
+  public void index_length_is_not_specified_on_big_varchar_columns_if_not_mysql() {
+    verifySql(new CreateIndexBuilder(new H2())
```

Comment:
you could write this test to loop on all supported DB but MySQL
that so, the implementation of the test would really match the method's name
(here, we only test H2 when name implies "all but mysql")